src/query/service/src/interpreters/interpreter_copy_v2.rs

```diff
@@ -49,6 +54,108 @@ impl CopyInterpreterV2 {
     }
 
+    async fn filter_duplicate_files(
+        &self,
+        force: bool,
+        table_info: &StageTableInfo,
+        catalog_name: &str,
+        database_name: &str,
+        table_name: &str,
+        files: &[String],
+    ) -> Result<(u64, BTreeMap<String, TableCopiedFileInfo>)> {
+        let catalog = self.ctx.get_catalog(catalog_name)?;
+        let tenant = self.ctx.get_tenant();
+        let table = catalog
+            .get_table(&tenant, database_name, table_name)
+            .await?;
+        let table_id = table.get_id();
+        let req = GetTableCopiedFileReq {
+            table_id,
+            files: files.to_owned(),
+        };
+        let mut file_map = BTreeMap::new();
+
+        if !force {
+            // if force is false, copy only the files that unmatch to the meta copied files info.
+            let resp = catalog.get_table_copied_file_info(req).await?;
+            for file in files.iter() {
+                let stage_file = list_files(&self.ctx, &table_info.stage_info, file, "").await?;
+                if stage_file.is_empty() {
+                    continue;
+                }
+
+                let stage_file = &stage_file[0];
+
+                if let Some(file_info) = resp.file_info.get(file) {
+                    match &file_info.etag {
+                        Some(_etag) => {
+                            // No need to copy the file again if etag is_some and match.
+                            if stage_file.etag == file_info.etag {
+                                tracing::warn!("ignore copy file {:?} matched by etag", file);
+                                continue;
+                            }
+                        }
+                        None => {
+                            // etag is none, compare with content_length and last_modified.
+                            if file_info.content_length == stage_file.size
+                                && file_info.last_modified == Some(stage_file.last_modified)
+                            {
+                                tracing::warn!(
+                                    "ignore copy file {:?} matched by content_length and last_modified",
+                                    file
+                                );
+                                continue;
+                            }
+                        }
+                    }
+                }
+
+                // unmatch case: insert into file map for copy.
+                file_map.insert(file.clone(), TableCopiedFileInfo {
+                    etag: stage_file.etag.clone(),
+                    content_length: stage_file.size,
+                    last_modified: Some(stage_file.last_modified),
+                });
+            }
+        } else {
+            // if force is true, copy all the file.
+            for file in files.iter() {
+                let stage_file = list_files(&self.ctx, &table_info.stage_info, file, "").await?;
```

Comment:
Don't use `list_files` to get the file's metadata. We can implement a function like the following:

```rust
async fn stat_file(ctx: &Arc<QueryContext>, stage: &UserStageInfo, path: &str) -> Result<StageFile>`
```

In which we can use `op.object(path).metadata().await`.